lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -104,23 +105,121 @@ class PatchEntryPointMutate : public Patch {
     bool mustSpill;         // Whether this is an arg that must be spilled
   };
 
+  // User data usage for one user data node
+  struct UserDataNodeUsage {
+    unsigned entryArgIdx = 0;
+    unsigned dwordSize = 0; // Only used in pushConstOffsets
+    SmallVector<Instruction *, 4> users;
+  };
+
+  // Per-merged-shader-stage gathered user data usage information.
+  struct UserDataUsage {
+    // List of lgc.spill.table calls
+    UserDataNodeUsage spillTable;
+    // List of lgc.push.const calls. There is no direct attempt to unspill these; instead we attempt to
+    // unspill the pushConstOffsets loads.
+    UserDataNodeUsage pushConst;
+    // True means that we did not succeed in putting all loads into pushConstOffsets, so lgc.push.const
+    // calls must be kept.
+    bool pushConstSpill = false;
+    // Per-push-const-offset lists of loads from push const. We attempt to unspill these.
+    SmallVector<UserDataNodeUsage, 8> pushConstOffsets;
+    // Per-user-data-offset lists of lgc.root.descriptor calls
+    SmallVector<UserDataNodeUsage, 8> rootDescriptors;
+    // Per-descriptor-set lists of lgc.descriptor.set calls
+    SmallVector<UserDataNodeUsage, 8> descriptorSets;
+    // Minimum offset at which spill table is used.
+    unsigned spillUsage = UINT_MAX;
```

Comment:
Actually, the spill threshold is set to UINT16_MAX by internal compiler :) Not an issue, but this could make the dump info consistent without any differences.